src/backend/joanie/edx_imports/tasks/users.py

```diff
@@ -0,0 +1,127 @@
+"""Celery tasks for importing data from the OpenEdx database to the Joanie database."""
+from logging import getLogger
+
+from joanie.celery_app import app
+from joanie.core import models
+from joanie.edx_imports.edx_database import OpenEdxDB
+from joanie.edx_imports.utils import extract_language_code, format_date
+
+# pylint: disable=too-many-locals
+
+logger = getLogger(__name__)
+
+
+def import_users(batch_size=1000, offset=0, limit=0, dry_run=False):
+    """Import users from OpenEdx auth_user"""
+    db = OpenEdxDB()
+    users_count = db.get_users_count(offset, limit)
+    if dry_run:
+        logger.info("Dry run: no user will be imported")
+    logger.info("%s users to import by batch of %s", users_count, batch_size)
+
+    batch_count = 0
+    for current_user_index in range(0, users_count, batch_size):
+        batch_count += 1
+        start = current_user_index + offset
+        stop = current_user_index + batch_size
+        import_users_batch_task.delay(start=start, stop=stop, dry_run=dry_run)
+    logger.info("%s import users tasks launched", batch_count)
+
+
+@app.task(bind=True)
+def import_users_batch_task(self, **kwargs):
+    """
+    Task to import users from the OpenEdx database to the Joanie database.
+    """
+    logger.info("Starting Celery task, importing users...")
+    try:
+        report = import_users_batch(**kwargs)
+    except Exception as e:
+        logger.exception(e)
+        raise self.retry(exc=e) from e
+    logger.info("Done executing Celery importing users task...")
+    return report
+
+
+def import_users_batch(start, stop, dry_run=False):
+    """Batch import users from OpenEdx auth_user"""
+    db = OpenEdxDB()
+    report = {
+        "users": {
+            "created": 0,
+            "updated": 0,
+            "errors": 0,
+        }
+    }
+    users = db.get_users(start, stop)
+    users_to_create = []
+    users_to_update = []
+
+    for edx_user in users:
+        if edx_user.username == "admin":
+            continue
+        last_login = format_date(edx_user.last_login)
+        date_joined = format_date(edx_user.date_joined)
+        language_code = extract_language_code(edx_user)
+        try:
+            user = models.User.objects.get(username=edx_user.username)
+            user.email = edx_user.email
+            user.password = edx_user.password
```

Comment:
I don't think we want to import password. Furthermore, I don't doing this manner will work until the application SECRET_KEY is the same between OpenEdX and Joanie

Reply:
The idea was not to set a usable password, but to avoid having the same for all users.

Reply:
We can set this value to `!`. This is a magic value to set a password that will be never matched.

Reply:
Here it is the magic 🪄 

https://github.com/django/django/blob/cf107fe255dbc4e1619c0985e4becdd9cabe8235/django/contrib/auth/hashers.py#L22